tools-and-tests/tools/src/main/java/org/hiero/block/tools/days/subcommands/GcpLsCommand.java

```diff
@@ -0,0 +1,398 @@
+
+    @Option(
+            names = {"--min-node"},
+            description = "Minimum node account ID (default: 3)",
+            defaultValue = "3")
+    private int minNodeAccountId;
+
+    @Option(
+            names = {"--max-node"},
+            description = "Maximum node account ID (default: 37)",
+            defaultValue = "37")
+    private int maxNodeAccountId;
+
+    @Option(names = "--parallelism", description = "Thread pool size for parallel node queries", defaultValue = "8")
+    private int parallelism;
+
+    @Option(
+            names = {"-p", "--user-project"},
+            description = "GCP project to bill for requester-pays bucket access (default: from GCP_PROJECT_ID env var)")
+    private String userProject = DownloadConstants.GCP_PROJECT_ID;
+
+    /**
+     * GCP mainnet bucket helper; assumed to have:
+     *  - listHour(LocalDate date, int hour, boolean includeSidecars)
+     */
+    private MainNetBucket mainNetBucket;
+
+    @Override
+    public void run() {
+        try {
+            if (mainNetBucket == null) {
+                mainNetBucket = new MainNetBucket(
+                        cacheEnabled, cacheDir.toPath(), minNodeAccountId, maxNodeAccountId, userProject);
+            }
+            if (!skipLocalCompare && downloadedDaysDir == null) {
+                throw new IllegalArgumentException("DownloadedDaysDir has not been set");
+            }
+
+            validateInputs();
+
+            System.out.printf("Sampling %d (day, hour) slots between %s and %s%n", samples, startDate, endDate);
+
+            long daysRange = ChronoUnit.DAYS.between(startDate, endDate) + 1;
+
+            ExecutorService executor = Executors.newFixedThreadPool(parallelism);
+            try {
+                for (int i = 0; i < samples; i++) {
+                    long dayOffset = ThreadLocalRandom.current().nextLong(daysRange);
+                    LocalDate sampleDate = startDate.plusDays(dayOffset);
+                    int hour = ThreadLocalRandom.current().nextInt(24);
+
+                    handleSample(i + 1, sampleDate, hour, executor);
+                }
+            } finally {
+                executor.shutdownNow();
+            }
+
+        } catch (Exception e) {
+            System.err.println("Error during gcp-ls: " + e.getMessage());
+            e.printStackTrace(System.err);
+        }
+    }
+
+    private void handleSample(
+            final int sampleIndex, final LocalDate sampleDate, final int hour, final ExecutorService executor)
+            throws ExecutionException, InterruptedException {
+
+        System.out.printf("%n=== Sample %d: date=%s hour=%02d ===%n", sampleIndex, sampleDate, hour);
+        System.out.printf("%-8s %-10s %-10s %-10s %-8s%n", "Node", "gcpFiles", "localFiles", "diff", "status");
+
+        // submit node tasks in parallel
+        final List<Future<NodeResult>> futures = new ArrayList<>();
+        for (int node = 0; node < nodeCount; node++) {
```

Comment:
Not sure why you look node's here. Should do all nodes at once. A single GCP ls using org.hiero.block.tools.utils.gcp.MainNetBucket.listHour will include all nodes. That is one of the features of  MainNetBucket class.